kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -664,6 +675,23 @@ public List<DomainMetadata> getComputedDomainMetadatasToCommit() {
       }
 
       generateClusteringDomainMetadataIfNeeded();
```

Comment:
do we still need to invoke `generateClusteringDomainMetadataIfNeeded` if it is a replaceTable call?

Reply:
yes! we might be replacing with a clustered table